UI/obs-app.cpp

```diff
@@ -58,6 +58,16 @@
 #include <pthread.h>
 #endif
 
+#if !defined(_WIN32) && !defined(__APPLE__)
+#include <obs-nix-platform.h>
+#include <QX11Info>
+
+#ifdef ENABLE_WAYLAND
+#include <qpa/qplatformnativeinterface.h>
```

Comment:
This concerns me. Is there no way to do what you want to do without requiring private Qt headers? I feel like I've asked about this before but it's been a long time.

Reply:
Hm, I'm not aware of any replacement for that. This seems to be the [documented way](https://wayland.freedesktop.org/qt5.html) to retrieve the `wl_surface` from a window.

Reply:
Yea that documentation seems to be from the wayland site, not the Qt site, which is fine I suppose, but generally it requires the qtbase5-private-dev on debian-based, and that might cause problems for other distributions. If you feel confident that other distros won't have problems, then that's fine I suppose. I don't really mind it, but usually using private headers for any dependency is somewhat taboo because of potential for trouble compiling at best, ABI breakage at worst if they change the virtual function table. But that particular header seems to be a known thing that people tend to use, so ABI breakage is *probably* unlikely, and Qt5 is pretty much on its last legs anyway so it probably doesn't matter.

Anyway, I hit this roadblock at first, and the cmake files weren't able to detect whether the required file/package was available or not. It just errored instead. I think that ideally, we should get cmake to detect whether the required files or packages are installed to ensure others don't have the same issue.

And if the package/file isn't detected/available, it should probably still build, but throw a cmake warning and disable building of wayland stuff or something. I'm not sure.

Ideally, I just want to make the build process cleaner.

Reply:
Ah, I see what you mean. I'll have a look at making CMake fail when that particular header is not there (it's the only header used from the private set, and I think it's safe to assume it'll be the only one too.)

Reply:
Okay, I think this CMake change should work (haven't tested though). KDE folks pointed out [this commit](https://github.com/KDAB/GammaRay/commit/6c1fb7c43d4e4733db511bf9d82ef7255cdddc69) which does roughly the same, and I've replicated it here.

Reply:
Looks good, detects it properly now. One question I might have is should we make it automatically compile without wayland support if the package isn't detected? Or should it be an explicit cmake option? I can't really decide. I think it's fine as is, but some people may be confused at first when they find that they can't compile OBS on Linux anymore. Might be good to have some sort of cmake warning at least telling them how to disable wayland support or something, or maybe just have it compile normally warning that wayland support was automatically disabled. I'm not sure.

Reply:
Leaving it as a dependency unless they explicitly turn off support seems best to me. At least this way maintainers know they need to take action (be it removing functionality or adding dependencies).